DllImportGenerator/DllImportGenerator/Analyzers/ManualTypeMarshallingAnalyzer.cs

```diff
@@ -486,12 +486,23 @@ private void AnalyzeNativeMarshalerType(SymbolAnalysisContext context, ITypeSymb
                     IMethodSymbol getPinnableReferenceMethodToMention = getPinnableReferenceMethods.Managed ?? getPinnableReferenceMethods.Marshaler!;
 
                     context.ReportDiagnostic(
-                        nativeTypeDiagnosticsTargetSymbol.CreateDiagnostic(
+                        GetDiagnosticLocations(context, nativeTypeDiagnosticsTargetSymbol, nativeMarshalerAttributeData).CreateDiagnostic(
                             NativeTypeMustBePointerSizedRule,
                             nativeType.ToDisplayString(),
                             getPinnableReferenceMethodToMention.ContainingType.ToDisplayString()));
                 }
             }
+
+            private ImmutableArray<Location> GetDiagnosticLocations(SymbolAnalysisContext context, ISymbol targetSymbol, AttributeData marshallingAttribute)
+            {
+                // If we're using a compilation that references another compilation, the symbol locations can be in source in the wrong compilation,
+                // which can cause exceptions when reporting diagnostics. Make sure the symbol is defined in the current Compilation's source module before using its locations.
```

Comment:
'Otherwise, report the diagnostic on the marshalling attribute' or something similar?